api/controllers/file.py

```diff
@@ -0,0 +1,120 @@
+import re
+from uuid import uuid4
+from fastapi import UploadFile, status, HTTPException
+from botocore.exceptions import ClientError
+from mypy_boto3_s3 import S3Client
+from sqlalchemy.orm import Session
+from sqlalchemy.exc import SQLAlchemyError
+
+from app import schema as s
+from app import models as m
+
+from app.logger import log
+from config import config
+
+S3_UPLOAD_EXTRAS = {"ACL": "public-read-write"}
+
+RE_SPECIAL_CHARACTERS = "[^a-zA-Z0-9 \n\.]"
+
+
+CFG = config()
+
+
+def create_file(
+    file: UploadFile,
+    db: Session,
+    s3_client: S3Client,
+    extension: str,
+    file_type: s.FileType,
+    content_type_override: str | None = None,
+) -> m.File:
+    try:
+        if not file.filename:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="File name is required",
+            )
+        filename_without_special_characters = re.sub(RE_SPECIAL_CHARACTERS, "", file.filename)
+        filename_without_spaces = filename_without_special_characters.replace(" ", "_")
+        file_uuid = str(uuid4())
+        filename = f"{file_uuid}_{filename_without_spaces}"
+        short_name = filename.split(".")[0]
+
+        key = f"jobs/files/{short_name}" + f".{extension}"
```

Comment:
files can relate not only to jobs. Please extract `jobs/files/` in function arguments